lib/scanner/list-scanner/tests/test_list_scanner.c

```diff
@@ -175,6 +175,21 @@ Test(list_scanner, works_with_gstring_input)
     g_string_free(argv[i], TRUE);
 }
 
+Test(list_scanner, works_with_simple_input)
+{
+  list_scanner_input_value(list_scanner, "foo,bar,baz", -1);
+  assert_next_value_is("foo");
+  assert_next_value_is("bar");
+  assert_next_value_is("baz");
+  assert_no_more_tokens();
+
+  /* nonzero terminated */
+  list_scanner_input_value(list_scanner, "foo,bar,baz", 7);
```

Comment:
Adding more tests to specify behaviour in case of escaped/non-escaped strings?
`value` is assumed to be a list, so to know the rules of a valid list would be good.
It is specified with test `handles_double_quotes`
I.e. 
- `foo ,bar,baz`
- `",foo",bar,baz`